components/teams-service/storage/postgres/postgres.go

```diff
@@ -353,53 +353,33 @@ func (p *postgres) PurgeUserMembership(ctx context.Context, userID string) ([]uu
 // server code before we get here if the team in question was filtered, but if we call this
 // from a different context we should apply project filtering in this function as well.
 // Not doing it to save us a database call since it's not needed currently.
-
 func (p *postgres) AddUsers(ctx context.Context,
 	teamID uuid.UUID,
 	userIDs []string) (storage.Team, error) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	tx, err := p.db.BeginTx(ctx, nil)
-	if err != nil {
-		return storage.Team{}, p.processError(err)
+	if len(userIDs) == 0 {
+		return storage.Team{}, nil
 	}
 
-	doneSomething := false
-	for _, userID := range userIDs {
-		res, err := tx.ExecContext(ctx,
-			`INSERT INTO teams_users_associations (team_db_id, user_id, created_at)
-			SELECT db_id, $2, now()
-			FROM teams
-			WHERE id=$1`, teamID, userID)
-		if err != nil {
-			return storage.Team{}, p.processError(err)
-		}
-		num, err := res.RowsAffected()
-		if err != nil {
-			return storage.Team{}, p.processError(err)
-		}
-		if num > 0 {
-			doneSomething = true
-		}
-	}
-	var team storage.Team
-
-	if !doneSomething {
-		team, err = p.getTeam(ctx, tx, teamID)
-
-	} else {
-		team, err = p.touchTeam(ctx, tx, teamID)
-	}
-	if err != nil {
-		return storage.Team{}, p.processError(err)
-	}
-
-	err = tx.Commit()
+	var t storage.Team
+	err := p.db.QueryRowContext(ctx,
+		`WITH moved_rows AS (
```

Comment:
this moved_rows pattern is new 👀how does it work?

Reply:
`moved_rows` is just an arbitrary name for the rows in the associations table affected by the update, which we can use later in the query to update the second table `teams`. We used to have to do this in multiple steps in golang but now it's a single query

Reply:
For more info: https://www.postgresql.org/docs/9.6/queries-with.html